File: src/app/interfaces/book.ts
import { Author } from "./author";
import { Category } from "./category";
import { Publisher } from "./publisher";

export interface Book {
  id: number;
  price: string;
  title: string;
  category_id: number;
  publisher_id: number;
  created_at: Date;
  updated_at: Date;
  authors: Author[];
  category: Category;
  publisher: Publisher;
}

File: app/models/book.rb
# frozen_string_literal: true

class Book < ApplicationRecord
  belongs_to :publisher
  belongs_to :category

  has_many :book_authors, dependent: :destroy
  has_many :authors,      through: :book_authors

  validates :title,  presence: true
  validates :price,  presence: true

  validates :price,  numericality: { greater_than_or_equal_to: 0 }
end
